### PR TITLE
Drop ECR for testing builds

### PR DESCRIPTION
**Pull request by feanil**
This PR simplifies the build and CI for codejail to make it easire to understand the testing and make it easier to test on multiple versions of Ubuntu.

Once this lands, the codejail code will support working on both Ubuntu 20.04 and Ubuntu 22.04 as long as the instructions in the readme are followed for setup.  Especially with respect the sudo PAM config: https://github.com/openedx/codejail/pull/201/commits/47842d9a20a019aada8aa7a53b9d610bffe6698d

### CODE REVIEW

**Comment by feanil** [comment]
@MoisesGSalas this is about how far I've gotten, I think it makes sense to land this without 24.04 for now and then try to fix that as a separate PR in the future.  It looks like something has changed either with docker or with apparmor in that version causing it to not enforce the network limits.   Can you review this and let me know if you have any concerns.  The commit messages should generally explain things but please ask any questions you might have.

**Comment by MoisesGSalas** [comment]
@feanil, I will take a look later tomorrow. But I'm wondering if running  the tests directly in the runner without docker (thinking of this [network](https://github.com/openedx/codejail/pull/201/commits/6d268f5320bd2286467be07fba3779e46bcb1ef6) issue) would be easier/better?

I found this [repo](https://github.com/roddhjav/apparmor.d/blob/main/.github/workflows/main.yml) that seems to use apparmor just fine in the CI runner.

**Comment by feanil** [comment]
@MoisesGSalas that's something we can consider so that we can just test the profiles without the docker related confusion but since we use docker as a part of tutor, it's useful to also test inside containers so I'm of two minds on the issue.  For now let's focus on landing this for 22.04 and we can come back to 24.04 after the release.

**Review by arbrandes — APPROVED** [review]
Thanks for walking me through it!  I can't find anything to object to.

**Review by MoisesGSalas — COMMENTED** [review]
Just one small comment

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -8,44 +8,40 @@ on:
 jobs:
   codejail_ci:
     name: tests
-    runs-on: ubuntu-20.04
+    runs-on: ${{ matrix.os }}
     strategy:
+      fail-fast: false
```
[the remark this answers is not in the document]
Comment (feanil) [reply .github/workflows/ci.yml:13]: At first but given how few test shards there are, I think it makes sense to run them independently so I think we should keep it this way, but happy to change it back if you disagree.

[the remark this answers is not in the document]
Comment (MoisesGSalas) [reply .github/workflows/ci.yml:13]: Looking at it, it only takes 2min to run the whole thing so I'm fine keeping it.

**Pull request merged**